hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcClient.java

```diff
@@ -81,4 +99,43 @@ protected void closeInternal() {
       group.shutdownGracefully();
     }
   }
+
+  SslContext getSslContext() throws X509Exception, SSLException {
+    SslContext result = sslContextForClient.get();
+    if (result == null) {
+      result = X509Util.createSslContextForClient(conf);
+      if (!sslContextForClient.compareAndSet(null, result)) {
+        // lost the race, another thread already set the value
+        result = sslContextForClient.get();
+      }
+    }
+    return result;
+  }
+
+  private ByteBufAllocator getByteBufAllocator(Configuration conf) throws IOException {
```

Comment:
As another comment points out, we do not need to pass a ByteBufAllocator to NettyRpcConnection, so let's not include this change in this PR. We can file another issue to land this change.

Reply:
I remove it from this patch, but do you think it still makes sense to do it separately?

Reply:
Can do it separately. And also about whether to enable epoll on client side.